### DISCUSSION TITLE
Log for Function Calling

### DISCUSSION THREAD

**Discussion by jaindees** [langfuse/langfuse#1780] [Ideas]
### Describe the feature or potential improvement

When can we have logs for function calling feature (agent tools), by default getting logged when a langchain based agent is invoked? It helps to understand what all functions/tools passed to LLM along with their descriptions and entity definitions. If it is already there then please help point me to the documentation. Thanks for the wonderful tool!!!

**Comment by marcklingen** [comment]
Can you please share a notebook that helps reproduce this issue? The tools and functions are logged to LangFuse for most integrations.

**Reply by jaindees** [reply]
Hi Marc,

Got some time to put this simple example together where tools/function description getting passed to LLM call are not getting logged as part of that LLM call in Langfuse. This is very crucial information to understand as what all functions got passed along with chat history and user query to LLM. I believe this contributes to token/cost calculation too. If it is not getting accounted then we are little off on token calculation too and also gets difficult to debug as what all tools/functions passed to LLM. I have also attached a screenshot from Langsmith as how they are logging this information as part of the invocation parameters as what all passed to the OpenAI API call. Something similar would be super helpful, especially for agent designs.

![Langsmith_Agent_Trace_With_Tools_Definition](https://github.com/langfuse/langfuse/assets/12931809/dbf2cae0-5bda-4fcf-8442-9998fa0eb5cf)

Here is the code for the same (change the extension of the file to .py to make it work).
[langchain_agent_langfuse_example.txt](https://github.com/langfuse/langfuse/files/15095776/langchain_agent_langfuse_example.txt)

On a side note, we are big fan of Langfuse Open Source and we have adopted it for our internal application. We also wrote a blog for the same. You might find it interesting, putting a link [here](https://medium.com/@genai_cybage_software/observability-an-essential-block-for-gen-ai-solutions-052eff38399d) for your reference and maybe consider it sharing as a case study for the community. Thanks!!!

**Reply by jaindees** [reply]
Hi @marcklingen , wondering if you got some time to look into this. Thanks!!!

**Reply by marcklingen** [reply]
Sorry for slow response here, I was a bit occupied by launch week. Turning this into an issue as this needs to be fixed

**Reply by humbertaco** [reply]
Hi, I am also interested in this. Could you please share the issue so I can keep track of it? :) Thanks!!

**Reply by jaindees** [reply]
Hi @marcklingen, wondering if this got addressed already or if you can point to issue maybe. Let us know. Thanks!!!

**Reply by jennyma** [reply]
@jaindees Have you seen any progress here?
